components/admin_console/billing/billing_history.test.tsx

```diff
@@ -0,0 +1,145 @@
+                                metadata: {},
+                            },
+                        ],
+                    },
+                    in_1KIWNTI67GP2qpb4KjGj1KAy: {
+                        id: 'in_1KIWNTI67GP2qpb4KjGj1KAy',
+                        number: '87030375-0013',
+                        create_at: 1642330467000,
+                        total: 0,
+                        tax: 0,
+                        status: 'paid',
+                        description: '',
+                        period_start: 1642330466000,
+                        period_end: 1642330466000,
+                        subscription_id: 'sub_1KIWNTI67GP2qpb49XXiEscG',
+                        line_items: [
+                            {
+                                price_id: 'price_1JMAbZI67GP2qpb4ADjRJwYa',
+                                total: 0,
+                                quantity: 1,
+                                price_per_unit: 1000,
+                                description:
+                                    'Trial period for Cloud Professional',
+                                type: 'onpremise',
+                                metadata: {},
+                            },
+                        ],
+                    },
+                },
+            },
+        },
+        views: {},
+    };
+
+    const NO_INVOICES_LEGEND = 'All of your monthly payments will show here';
+
+    const mockStore = configureStore([thunk]);
+    const store = mockStore(state);
+
+    test('should match snapshot', () => {
+        const wrapper = shallow(
+            <Provider store={store}>
+                <BillingHistory/>
+            </Provider>,
+        );
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('Billing history section shows template when no invoices have been emmitted yet', () => {
+        const noBillingHistoryState = {
+            ...state,
+            entities: {...state.entities, cloud: {invoices: {}}},
+        };
+        const storeNoBillingHistory = mockStore(noBillingHistoryState);
+        const wrapper = mountWithIntl(
+            <Provider store={storeNoBillingHistory}>
+                <BillingHistory/>
+            </Provider>,
+        );
+
+        const legend = wrapper.find(
+            '.BillingHistory__cardHeaderText-bottom span',
+        );
+        expect(legend.text()).toBe(NO_INVOICES_LEGEND);
+    });
+
+    test('Billing history section shows two invoices to download', () => {
+        const wrapper = mountWithIntl(
+            <Provider store={store}>
+                <BillingHistory/>
+            </Provider>,
+        );
+
+        const invoiceTableRows = wrapper.find('table.BillingHistory__table tr.BillingHistory__table-row');
+
+        expect(invoiceTableRows.length).toBe(2);
+    });
+
+    test('Billing history section download button has the target property set as _self so it works well in desktop app', () => {
```

Comment:
Added unit tests to cover this scenario from a future regression by again changing the target property value